src/ui/hooks/tests.ts

```diff
@@ -212,28 +213,31 @@ function convertTest(test: GetTest_node_Workspace_tests | undefined) {
 }
 
 export function useGetTestRunsForWorkspace(workspaceId: WorkspaceId): {
-  testRuns: TestRun[] | null;
+  testRuns: TestRun[];
   loading: boolean;
 } {
+  // TODO [FE-1530] Pagination test runs from GraphQL
   const { data, loading } = useQuery<GetTestsRunsForWorkspace, GetTestsRunsForWorkspaceVariables>(
     GET_TEST_RUNS_FOR_WORKSPACE,
     {
       variables: { workspaceId },
     }
   );
 
-  if (loading || !data?.node) {
-    return { testRuns: null, loading };
-  }
-  assert("testRuns" in data.node, "No testRuns in GetTestsRun response");
+  return useMemo(() => {
```

Comment:
⚠️ Important: Memoize the new Array we create (via `.map`) so we don't cause lots of unnecessary render work for React